spatio_temporal_voxel_layer/src/measurement_buffer.cpp

```diff
@@ -137,44 +137,89 @@ void MeasurementBuffer::BufferROSCloud(
       return;
     }
 
```

Comment:
For this block: move TF lookups into the `switch` statement so that we only lookup the few required. I don't want excess TF lookups for situations they're not used (TF lookups can add latency).

Reply:
Plus if z reference frame is an empty string, that'll throw exceptions in TF if its attempted to look up